pkg/remote/aws/ecr_repository_policy_enumerator.go

```diff
@@ -0,0 +1,51 @@
+package aws
+
+import (
+	"github.com/snyk/driftctl/pkg/remote/aws/repository"
+	remoteerror "github.com/snyk/driftctl/pkg/remote/error"
+	"github.com/snyk/driftctl/pkg/resource"
+	"github.com/snyk/driftctl/pkg/resource/aws"
+)
+
+type ECRRepositoryPolicyEnumerator struct {
+	repository repository.ECRRepository
+	factory    resource.ResourceFactory
+}
+
+func NewECRRepositoryPolicyEnumerator(repo repository.ECRRepository, factory resource.ResourceFactory) *ECRRepositoryPolicyEnumerator {
+	return &ECRRepositoryPolicyEnumerator{
+		repository: repo,
+		factory:    factory,
+	}
+}
+
+func (e *ECRRepositoryPolicyEnumerator) SupportedType() resource.ResourceType {
+	return aws.AwsEcrRepositoryPolicyResourceType
+}
+
+func (e *ECRRepositoryPolicyEnumerator) Enumerate() ([]*resource.Resource, error) {
+	repos, err := e.repository.ListAllRepositories()
+	if err != nil {
+		return nil, remoteerror.NewResourceListingErrorWithType(err, string(e.SupportedType()), aws.AwsEcrRepositoryPolicyResourceType)
```

Comment:
```suggestion
		return nil, remoteerror.NewResourceListingErrorWithType(err, string(e.SupportedType()), aws.AwsEcrRepositoryResourceType)
```
You fail to read the ecr repository policies because it's impossible to list ecr repositories. The error constructeur first argument is the enumerator type and the second one is the one unable to list, in this case they are different.